### PR TITLE
Remove unsafe `eq` instance `MutList`

### PR DESCRIPTION
**Pull request by ninaandrup**
#2642

### CODE REVIEW

File: main/test/ca/uwaterloo/flix/library/TestChain.flix [hunk 1]
```diff
@@ -375,16 +375,16 @@ namespace TestChain {
 
     @test
     def repeat01(): Bool = Chain.repeat(-1, 1) == Chain.empty()
-    
+
```
Comment (ninaandrup) [inline main/test/ca/uwaterloo/flix/library/TestChain.flix:378]: I think maybe github deleted all the indents with no line

Comment (magnus-madsen) [reply main/test/ca/uwaterloo/flix/library/TestChain.flix:378]: OK, that is fine.

**Comment by magnus-madsen** [comment]
No comments, so merged!

**Comment by magnus-madsen** [comment]
By the way, we forgot to rename the PR to follow the new scheme. https://github.com/flix/flix/blob/master/docs/PULLREQUEST.md

oh well. next time :)

**Pull request merged**